### ISSUE TITLE
Nimble fails to install package

### ISSUE THREAD

**Issue by futureisfoss** [nim-lang/langserver#46]
Trying to install the `nimlangserver` package ends with this error:
```
Downloading https://github.com/nim-lang/Nim.git using git
sh: line 1: ./build_all.sh: Permission denied
stack trace: (most recent call last)
/tmp/nimblecache-3864897998/nimscriptapi_1088818555.nim(222, 29)
/tmp/nimble_24447/githubcom_nimlangNimgit_2.1.1_940b1607b8459b4b7b7e20d316bec95c8de85809/nim.nimble(17, 7) installBefore
/home/rokosun/.choosenim/toolchains/nim-2.0.0/lib/system/nimscript.nim(265, 7) exec
/home/rokosun/.choosenim/toolchains/nim-2.0.0/lib/system/nimscript.nim(265, 7) Error: unhandled exception: FAILED: ./build_all.sh [OSError]
       Tip: 1080 messages have been suppressed, use --verbose to show them.
nimscriptwrapper.nim(160) execScript

    Error:  Exception raised during nimble script execution
```
Full output below:
```
~ >>> nimble install nimlangserver
Downloading https://github.com/nim-lang/langserver using git
Downloading https://github.com/status-im/nim-unittest2.git using git
 Installing unittest2@0.0.7
  Success:  unittest2 installed successfully.
Downloading https://github.com/status-im/nim-testutils using git
 Installing testutils@0.5.0
   Building testutils/ntu using c backend
/tmp/nimble_24447/githubcom_statusimnimtestutils_0.5.0_dfc4c1b39f9ded9baf6365014de2b4bfb4dafc34/testutils/spec.nim(160, 7) Warning: Using an unnamed break in a block is deprecated; Use a named block with a named break instead [UnnamedBreak]
/tmp/nimble_24447/githubcom_statusimnimtestutils_0.5.0_dfc4c1b39f9ded9baf6365014de2b4bfb4dafc34/testutils/helpers.nim(67, 32) Warning: Deprecated since 1.5; TaintedString is deprecated [Deprecated]
/tmp/nimble_24447/githubcom_statusimnimtestutils_0.5.0_dfc4c1b39f9ded9baf6365014de2b4bfb4dafc34/ntu.nim(61, 15) template/generic instantiation of `styledEcho` from here
/home/rokosun/.choosenim/toolchains/nim-2.0.0/lib/pure/terminal.nim(830, 9) template/generic instantiation of `styledWriteLine` from here
/tmp/nimble_24447/githubcom_statusimnimtestutils_0.5.0_dfc4c1b39f9ded9baf6365014de2b4bfb4dafc34/ntu.nim(62, 33) Warning: binary is deprecated [Deprecated]
/tmp/nimble_24447/githubcom_statusimnimtestutils_0.5.0_dfc4c1b39f9ded9baf6365014de2b4bfb4dafc34/ntu.nim(167, 7) Warning: Using an unnamed break in a block is deprecated; Use a named block with a named break instead [UnnamedBreak]
/tmp/nimble_24447/githubcom_statusimnimtestutils_0.5.0_dfc4c1b39f9ded9baf6365014de2b4bfb4dafc34/ntu.nim(193, 11) Warning: Using an unnamed break in a block is deprecated; Use a named block with a named break instead [UnnamedBreak]
/tmp/nimble_24447/githubcom_statusimnimtestutils_0.5.0_dfc4c1b39f9ded9baf6365014de2b4bfb4dafc34/ntu.nim(203, 13) Warning: Using an unnamed break in a block is deprecated; Use a named block with a named break instead [UnnamedBreak]
/tmp/nimble_24447/githubcom_statusimnimtestutils_0.5.0_dfc4c1b39f9ded9baf6365014de2b4bfb4dafc34/ntu.nim(212, 11) Warning: Using an unnamed break in a block is deprecated; Use a named block with a named break instead [UnnamedBreak]
/tmp/nimble_24447/githubcom_statusimnimtestutils_0.5.0_dfc4c1b39f9ded9baf6365014de2b4bfb4dafc34/ntu.nim(237, 16) Warning: binary is deprecated [Deprecated]
/tmp/nimble_24447/githubcom_statusimnimtestutils_0.5.0_dfc4c1b39f9ded9baf6365014de2b4bfb4dafc34/testutils/config.nim(125, 13) Warning: Potential object case transition, instantiate new object instead [CaseTransition]
  Success:  testutils installed successfully.
Downloading https://github.com/status-im/nim-stew using git
 Installing stew@0.1.0
  Success:  stew installed successfully.
Downloading https://github.com/cheatfate/nimcrypto using git
 Installing nimcrypto@0.5.4
  Success:  nimcrypto installed successfully.
Downloading https://github.com/status-im/nim-stint using git
 Installing stint@0.0.1
  Success:  stint installed successfully.
Downloading https://github.com/status-im/nim-bearssl using git
 Installing bearssl@0.2.0
  Success:  bearssl installed successfully.
Downloading https://github.com/status-im/nim-http-utils using git
 Installing httputils@0.3.0
  Success:  httputils installed successfully.
Downloading https://github.com/status-im/nim-chronos using git
 Installing chronos@3.2.0
  Success:  chronos installed successfully.
Downloading https://github.com/status-im/nim-faststreams using git
 Installing faststreams@0.3.0
  Success:  faststreams installed successfully.
Downloading https://github.com/status-im/nim-serialization using git
 Installing serialization@0.2.0
  Success:  serialization installed successfully.
Downloading https://github.com/status-im/nim-json-serialization using git
 Installing json_serialization@0.1.5
  Success:  json_serialization installed successfully.
Downloading https://github.com/status-im/nim-chronicles using git
 Installing chronicles@0.10.2
  Success:  chronicles installed successfully.
Downloading https://github.com/status-im/news using git
 Installing news@0.5
  Success:  news installed successfully.
Downloading https://github.com/status-im/nim-zlib using git
 Installing zlib@0.1.0
  Success:  zlib installed successfully.
Downloading https://github.com/status-im/nim-websock using git
 Installing websock@0.1.0
  Success:  websock installed successfully.
Downloading https://github.com/yyoncho/asynctools using git
 Installing asynctools@0.1.1
  Success:  asynctools installed successfully.
Downloading https://github.com/yyoncho/nim-json-rpc using git
 Installing json_rpc@0.0.2
  Success:  json_rpc installed successfully.
Downloading https://github.com/zevv/with using git
 Installing with@0.4.0
  Success:  with installed successfully.
Downloading https://github.com/nim-lang/Nim.git using git
sh: line 1: ./build_all.sh: Permission denied
stack trace: (most recent call last)
/tmp/nimblecache-3864897998/nimscriptapi_1088818555.nim(222, 29)
/tmp/nimble_24447/githubcom_nimlangNimgit_2.1.1_940b1607b8459b4b7b7e20d316bec95c8de85809/nim.nimble(17, 7) installBefore
/home/rokosun/.choosenim/toolchains/nim-2.0.0/lib/system/nimscript.nim(265, 7) exec
/home/rokosun/.choosenim/toolchains/nim-2.0.0/lib/system/nimscript.nim(265, 7) Error: unhandled exception: FAILED: ./build_all.sh [OSError]
       Tip: 1080 messages have been suppressed, use --verbose to show them.
nimscriptwrapper.nim(160) execScript
    Error:  Exception raised during nimble script execution
```
Debug output:
```
~ >>> nimble --debug install nimlangserver
     Info:  Nimble data file "/home/rokosun/.nimble/nimbledata2.json" has been loaded.
    Reading official package list
Downloading https://github.com/nim-lang/langserver using git
  Executing git ls-remote --tags https://github.com/nim-lang/langserver
    Cloning latest tagged version: v1.0.0
  Executing git clone --config core.autocrlf=false --config core.eol=lf --recursive --depth 1 -b v1.0.0 https://github.com/nim-lang/langserver /tmp/nimble_2752/githubcom_nimlanglangserver
  Executing git -C /tmp/nimble_2752/githubcom_nimlanglangserver rev-parse HEAD
     Debug: Executing /home/rokosun/.nimble/bin/nim e --hints:off --verbosity:0 --colors:on  /tmp/nimblecache-3864897998/nimlangserver_1155338580/nimlangserver.nims /tmp/nimble_2752/githubcom_nimlanglangserver/nimlangserver.nimble /tmp/nimble_2752.out printPkgInfo
   Warning: Not removing temporary path because of debug verbosity: /tmp/nimble_2752.out
  Executing git -C /tmp/nimble_2752/githubcom_nimlanglangserver ls-files
  Executing git -C /tmp/nimble_2752/githubcom_nimlanglangserver rev-parse HEAD
  Executing git -C /tmp/nimble_2752/githubcom_nimlanglangserver rev-parse --abbrev-ref --symbolic-full-name @{u}
  Executing git -C /tmp/nimble_2752/githubcom_nimlanglangserver branch --show-current
  Executing git -C /tmp/nimble_2752/githubcom_nimlanglangserver remote get-url  origin
Downloading https://github.com/nim-lang/Nim.git using git
    Cloning revision: 940b1607b8459b4b7b7e20d316bec95c8de85809
  Executing git -C /tmp/nimble_2752/githubcom_nimlangNimgit_2.1.1_940b1607b8459b4b7b7e20d316bec95c8de85809 init
  Executing git -C /tmp/nimble_2752/githubcom_nimlangNimgit_2.1.1_940b1607b8459b4b7b7e20d316bec95c8de85809 config core.autocrlf false
  Executing git -C /tmp/nimble_2752/githubcom_nimlangNimgit_2.1.1_940b1607b8459b4b7b7e20d316bec95c8de85809 remote add origin https://github.com/nim-lang/Nim.git
  Executing git -C /tmp/nimble_2752/githubcom_nimlangNimgit_2.1.1_940b1607b8459b4b7b7e20d316bec95c8de85809 fetch --depth 1 origin 940b1607b8459b4b7b7e20d316bec95c8de85809
  Executing git -C /tmp/nimble_2752/githubcom_nimlangNimgit_2.1.1_940b1607b8459b4b7b7e20d316bec95c8de85809 reset --hard FETCH_HEAD
  Executing git -C /tmp/nimble_2752/githubcom_nimlangNimgit_2.1.1_940b1607b8459b4b7b7e20d316bec95c8de85809 submodule update --init --recursive --depth 1
     Debug: Executing /home/rokosun/.nimble/bin/nim e --hints:off --verbosity:0 --colors:on  /tmp/nimblecache-3864897998/nim_2528002364/nim.nims /tmp/nimble_2752/githubcom_nimlangNimgit_2.1.1_940b1607b8459b4b7b7e20d316bec95c8de85809/nim.nimble /tmp/nimble_2752.out printPkgInfo
  Executing git -C /tmp/nimble_2752/githubcom_nimlangNimgit_2.1.1_940b1607b8459b4b7b7e20d316bec95c8de85809 ls-files
  Executing git -C /tmp/nimble_2752/githubcom_nimlangNimgit_2.1.1_940b1607b8459b4b7b7e20d316bec95c8de85809 rev-parse HEAD
  Executing git -C /tmp/nimble_2752/githubcom_nimlangNimgit_2.1.1_940b1607b8459b4b7b7e20d316bec95c8de85809 rev-parse --abbrev-ref --symbolic-full-name @{u}
  Executing git -C /tmp/nimble_2752/githubcom_nimlangNimgit_2.1.1_940b1607b8459b4b7b7e20d316bec95c8de85809 branch --show-current
  Executing git -C /tmp/nimble_2752/githubcom_nimlangNimgit_2.1.1_940b1607b8459b4b7b7e20d316bec95c8de85809 remote get-url  origin
  Executing git -C /tmp/nimble_2752/githubcom_nimlangNimgit_2.1.1_940b1607b8459b4b7b7e20d316bec95c8de85809 ls-files
 Attempting to execute hook installBefore in /tmp/nimble_2752/githubcom_nimlangNimgit_2.1.1_940b1607b8459b4b7b7e20d316bec95c8de85809/nim.nimble
     Debug: Executing /home/rokosun/.nimble/bin/nim e --hints:off --verbosity:0 --colors:on  /tmp/nimblecache-3864897998/nim_2528002364/nim.nims /tmp/nimble_2752/githubcom_nimlangNimgit_2.1.1_940b1607b8459b4b7b7e20d316bec95c8de85809/nim.nimble /tmp/nimble_2752.out installBefore
sh: line 1: ./build_all.sh: Permission denied
stack trace: (most recent call last)
/tmp/nimblecache-3864897998/nimscriptapi_1088818555.nim(222, 29)
/tmp/nimble_2752/githubcom_nimlangNimgit_2.1.1_940b1607b8459b4b7b7e20d316bec95c8de85809/nim.nimble(17, 7) installBefore
/home/rokosun/.choosenim/toolchains/nim-2.0.0/lib/system/nimscript.nim(265, 7) exec
/home/rokosun/.choosenim/toolchains/nim-2.0.0/lib/system/nimscript.nim(265, 7) Error: unhandled exception: FAILED: ./build_all.sh [OSError]
nimscriptwrapper.nim(160) execScript
    Error:  Exception raised during nimble script execution
   Warning: Not removing temporary path because of debug verbosity: /tmp/nimble_2752
     Info:  Nimble data file "/home/rokosun/.nimble/nimbledata2.json" has been saved.
```

More info:

I use [Artix linux](https://artixlinux.org/) and have installed the [choosenim-bin](https://aur.archlinux.org/packages/choosenim-bin) package from the AUR, after which I ran `choosenim stable` command to install the latest stable version of Nim (which is currently `Nim 2.0.0`). And of course I have `/home/rokosun/.nimble/bin` added to my `$PATH` variable as well.

**Comment by yyoncho** [comment]
can you try with the latest version of nimble? You may download one from there: https://github.com/nim-lang/nimble/releases/tag/latest

**Comment by futureisfoss** [comment]
UPDATE: I did some more troubleshooting and figured out the issue, it had to do with the way I mounted my `/tmp` directory to RAM.

On most linux distros systemd would automatically mount `/tmp` to RAM, but since I use a non-systemd distro I had to do it manually. I had previously followed the instructions on [this askUbuntu post](https://askubuntu.com/questions/173094/how-can-i-use-ram-storage-for-the-tmp-directory-and-how-to-set-a-maximum-amount) to make my `/tmp` directory use RAM instead of hard disk by adding this to my `/etc/fstab`:

```
tmpfs /tmp tmpfs defaults,noatime,nosuid,nodev,noexec,mode=1777,size=512M 0 0
```

I think the `noxec` part is what caused the permission denied error here. So I changed that fstab entry by replacing it with a different one I got [from ArchWiki](https://wiki.archlinux.org/title/Tmpfs):

```
tmpfs   /tmp         tmpfs   rw,nodev,nosuid,size=2G          0  0
```

Doing this and restarting my laptop actually fixed the permission denied error I got before, however this also created a problem because when I ran `nimble install nimlangserver` again it started putting a lot of files in `/tmp` filling up my RAM - I never expected it to reach the 2GB limit but unfortunately it did and my laptop froze as a result (it's an old machine with only 4GB RAM), I had to force power-off my laptop to get back to a working system. After that I removed the fstab entry entirely so that `/tmp` directory won't be using RAM anymore, restarted my laptop and now nimble was able to successfully install nimlangserver. This temporary workaround solved my problem but I had to go through a lot of troubleshooting before I figured it out, here are some of my key takeaways:

1. The `--debug` option was very useful because it stopped nimble from cleaning out the files it had put on my `/tmp` directory, and that's what actually helped me to figure out that the permission denied error was due to my fstab entry. Before I was using the `--verbose` option instead without realizing that there was a `--debug` option which is actually more verbose than the `--verbose` option.
2. When I was trying to install `nimlangserver` it seemed like nimble had to compile the entire nim package, correct me if I'm wrong here but I don't understand why this was necessary when I already had them installed via choosenim.
3. The files that nimble had put inside my `/tmp` directory exceeded the size of 2GB (even tho its temporary), this is not good for machines with a lower RAM size such as mine, considering that I think it might be better and more resilient to use the `~/.cache` directory instead.
